src/js/lib/search.js

```diff
@@ -106,44 +106,78 @@ export const buildSearchQuery = ({ text, from, size, sort }) => {
     builder.sort(field, option)
   }
 
-  const queryType = isDoubleQuoted(text) ? "query_string" : "multi_match"
-  const textQuery = emptyOrNil(text) ?
-    {} :
-    {
-      should: [
-        {
-          [queryType]: {
-            query:  text,
-            fields: searchFields(LR_TYPE_COURSE)
-          }
-        }
-      ]
-    }
+  for (const type of [LR_TYPE_COURSE]) {
+    const queryType = isDoubleQuoted(text) ? "query_string" : "multi_match"
+    const textQuery = emptyOrNil(text) ?
+      {} :
+      {
+        should: [
+          {
+            [queryType]: {
+              query:  text,
+              fields: searchFields(type)
+            }
+          },
+          type === LR_TYPE_COURSE ?
+            [
+              {
+                nested: {
+                  path:  "runs",
+                  query: {
+                    [queryType]: {
+                      query:  text,
+                      fields: RESOURCE_QUERY_NESTED_FIELDS
+                    }
+                  }
+                }
+              },
+              {
+                has_child: {
+                  type:  "resourcefile",
+                  query: {
+                    [queryType]: {
+                      query:  text,
+                      fields: ["content", "title", "short_description"]
+                    }
+                  },
+                  score_mode: "avg"
+                }
+              }
+            ] :
+            null
+        ]
+          .flat()
+          .filter(clause => clause !== null)
+      }
 
-  // buildFacetSubQuery
-  const facets = new Map([
-    ["type", [LR_TYPE_COURSE]],
-    ["offered_by", [OCW_PLATFORM]]
-  ])
-  const facetClauses = buildFacetSubQuery(facets, builder)
+    // buildFacetSubQuery
+    const facets = new Map([
+      ["audience", []],
+      ["certification", []],
+      ["type", [LR_TYPE_COURSE]],
+      ["offered_by", [OCW_PLATFORM]],
+      ["topics", []]
```

Comment:
The new facets above don't seem to do anything yet, but they don't seem to hurt either.

Reply:
yeah I pulled them in because I'm planning to add facets as soon as this PR merges: https://github.com/mitodl/course-search-utils/pull/1